src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1207,34 +1147,50 @@ private void ExecuteAndCatchIfThrows_NoLock<TArg>(DiagnosticAnalyzer analyzer, A
                 cancellationToken.ThrowIfCancellationRequested();
                 analyze(argument);
             }
-            catch (Exception e) when (ExceptionFilter(e, cancellationToken))
```

Comment:
>when (ExceptionFilter(e, cancellationToken)) [](http://example.com/codeflow?start=32&length=44)

What is the impact of moving the `ExceptionFilter()` call into the `catch` body? Are we losing the original stack when the exception should not have been caught? And if so, is that important?

Reply:
Ah good point - let me adjust the code here so we re-throw within the catch body itself to preserve the original exception callstack.